### ISSUE TITLE
bug: GcdsButton :aria-expanded reactive binding fails to update in Shadow DOM on initial trigger

### ISSUE THREAD

**Issue by kluu13** [cds-snc/gcds-components#937]
### Prerequisites | Prérequis

- [x] I have read the [Contributing Guidelines](https://github.com/cds-snc/gcds-components/blob/main/CONTRIBUTING.md). | J’ai lu les [Lignes directrices sur les contributions](https://github.com/cds-snc/gcds-components/blob/main/CONTRIBUTING.md).
- [x] I agree to follow the [Code of Conduct](https://github.com/cds-snc/gcds-components/blob/main/CODE_OF_CONDUCT.md). | J’accepte de me conformer au [Code de conduite](https://github.com/cds-snc/gcds-components/blob/main/CODE_OF_CONDUCT.md).
- [x] I have searched for [existing issues](https://github.com/cds-snc/gcds-components/issues) that already report this problem, without success. | J’ai vérifié parmi [les tickets (issues) existants](https://github.com/cds-snc/gcds-components/issues) que ce sujet n’avait pas déjà été soumis.

### GC Design System Components Package and Version | Paquet et version des composants de Système de design GC

gcds-components-vue@0.40.0

### Current Behavior | Comportement observé

The `:aria-expanded` reactive attribute binding on the `GcdsButton` component doesn't update properly on initial trigger, causing the expanded state to be inverted on subsequent interactions. The same code works correctly with a native `<button>` element.

### Expected Behavior | Comportement attendu

`aria-expanded` should update each time that the binded reactive property changes.

### System Info | Information sur le système

```shell
System: Windows 11
Browsers: Chrome, Edge
Framework & Tooling: Vue (3.5.17), Vite (6.0.0)
```

### Steps to Reproduce | Étapes pour reproduire le bogue

1. Create a `GcdsButton` with :`aria-expanded` binded to a reactive boolean property.
2. Add a click handler that toggles the boolean value.
3. Click the button and, using browser Dev Tools, inspect the `aria-expanded` attribute of the button element in the shadow DOM.
4. Observe that `aria-expanded` doesn't update on first click.
5. Click again and see that `aria-expanded` now shows the inverted state from step 3.
